src/WebJobs.Script.WebHost/Models/HostAssignmentContext.cs

```diff
@@ -129,6 +133,23 @@ public void ApplyAppSettings(IEnvironment environment)
                     environment.SetEnvironmentVariable(EnvironmentSettingNames.CorsAllowedOrigins, allowedOrigins);
                 }
             }
+
+            if (EasyAuthSettings != null)
+            {
+                // App settings take precedence over site config for easy auth enabled.
+                if (environment.GetEnvironmentVariable(EnvironmentSettingNames.EasyAuthEnabled) == null)
+                {
+                    logger.LogWarning($"ApplyAppSettings is adding {EnvironmentSettingNames.EasyAuthEnabled} = {EasyAuthSettings.SiteAuthEnabled.ToString()}");
+                    environment.SetEnvironmentVariable(EnvironmentSettingNames.EasyAuthEnabled, EasyAuthSettings.SiteAuthEnabled.ToString());
+                }
+                else
+                {
```

Comment:
Simplify. Assign defaults first i.e remove else block, move code above else block.  Add if block and override settings if needed. Also, take a look at the logging statements to see if they can be simiplified.